### PR TITLE
NOISSUE - Fix opcua-adapter events decode

### PR DESCRIPTION
**Pull request by manuio**
With an existing event processing implementation, whenever the processor receives a new event with some unexpected format it will break. This implementation should be able to handle unexpected metadata formats in a better way.

### CODE REVIEW

File: opcua/gopcua/sub.go [hunk 1]
```diff
@@ -53,16 +53,16 @@ func (b client) Subscribe(cfg opcua.Config) error {
 	}
 
 	c := opcuaGopcua.NewClient(ep.EndpointURL, opts...)
-	if errC := c.Connect(b.ctx); err != nil {
-		b.logger.Error(errC.Error())
+	if errc := c.Connect(b.ctx); errc != nil {
```
Comment (nmarcetic) [inline opcua/gopcua/sub.go]: Why not just follow err propagation? Why new var errc and errs?

File: opcua/redis/streams.go [hunk 2]
```diff
@@ -89,12 +93,12 @@ func (es eventStore) Subscribe(subject string) error {
 				}
 				err = es.handleCreateThing(cte)
 			case thingUpdate:
-				ute, derr := decodeUpdateThing(event)
+				ute, derr := decodeCreateThing(event)
```
Comment (nmarcetic) [inline opcua/redis/streams.go]: derr  ? why not err ?

**Review by nmarcetic — DISMISSED** [review]

File: opcua/gopcua/sub.go (hunk 1 above)
Comment (drasko) [reply opcua/gopcua/sub.go]: Indeed, `err` is a special var in Go that can be re-assigned with `:=`

File: opcua/redis/streams.go [hunk 3]
```diff
@@ -189,39 +176,23 @@ func decodeCreateChannel(event map[string]interface{}) (createChannelEvent, erro
...
-func decodeUpdateChannel(event map[string]interface{}) (updateChannelEvent, error) {
-	strmeta := read(event, "metadata", "{}")
-	var metadata map[string]channelMetadata
-	if err := json.Unmarshal([]byte(strmeta), &metadata); err != nil {
-		return updateChannelEvent{}, err
-	}
-
-	uce := updateChannelEvent{
-		id: read(event, "id", ""),
+	om, ok := m.(map[string]interface{})
```
Comment (drasko) [inline opcua/redis/streams.go]: What does `om` stand for? Without going into the usbject, var name is a bit confusing.

**Review by drasko — CHANGES_REQUESTED** [review]

File: opcua/gopcua/sub.go [hunk 4]
```diff
@@ -53,18 +53,19 @@ func (b client) Subscribe(cfg opcua.Config) error {
 	}
 
 	c := opcuaGopcua.NewClient(ep.EndpointURL, opts...)
-	if errC := c.Connect(b.ctx); err != nil {
-		b.logger.Error(errC.Error())
+	if err := c.Connect(b.ctx); err != nil {
+		return fmt.Errorf("Failed to connect: %s", err)
```
Comment (dborovcanin) [inline opcua/gopcua/sub.go]: You can use our `errors` package here, something like: `errors.Wrap(ErrFailedConn, err)`.

**Review by drasko — DISMISSED** [review]
LGTM

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Review by nmarcetic — APPROVED** [review]

**Pull request merged**